system-test/bigquery.ts

```diff
@@ -676,6 +677,79 @@ describe('BigQuery', () => {
     });
   });
 
+  describe('BigQuery/Routine', () => {
+    after(async () => {
+      const [routines] = await dataset.getRoutines();
+      return Promise.all(routines.map(routine => routine.delete()));
+    });
+
+    it('should create a routine via insert', () => {
+      return dataset.createRoutine('my_routine', {
+        arguments: [
+          {
+            name: 'x',
+            dataType: {
+              typeKind: 'INT64',
+            },
+          },
+        ],
+        definitionBody: 'x * 3',
+        routineType: 'SCALAR_FUNCTION',
+        returnType: {
+          typeKind: 'INT64',
+        },
+      });
+    });
+
+    it('should create a routine via DDL', () => {
+      const routineId = `${bigquery.projectId}.${dataset.id}.my_ddl_routine`;
+
+      return bigquery.query(`
+        CREATE FUNCTION \`${routineId}\`(
+          arr ARRAY<STRUCT<name STRING, val INT64>>
+        ) AS (
+          (SELECT SUM(IF(elem.name = "foo",elem.val,null)) FROM UNNEST(arr) AS elem)
+        )
+      `);
+    });
+
+    it('should list all the routines', async () => {
+      const [routines] = await dataset.getRoutines();
+      assert.strictEqual(routines.length, 2);
+      assert.ok(routines[0] instanceof Routine);
+    });
+
+    it('should get the routines as a stream', done => {
+      const routines: Routine[] = [];
+
+      dataset
+        .getRoutinesStream()
+        .on('error', done)
+        .on('data', routine => {
+          routines.push(routine);
+        })
+        .on('end', () => {
+          assert.strictEqual(routines.length, 2);
+          assert.ok(routines[0] instanceof Routine);
+          done();
+        });
+    });
+
+    it('should check to see if a routine exists', async () => {
+      const routine = dataset.routine('my_routine');
```

Comment:
Same idea as the other one, where we should create the routine we're checking the existence on in this same test.

Or... create the routine in the `before`, and re-use the object throughout the tests.